File: app/src/main/java/com/callmangement/EHR/ehrActivities/ViewCampDetailsActivity.java
package com.callmangement.EHR.ehrActivities;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import com.callmangement.R;
import androidx.recyclerview.widget.GridLayoutManager;

import com.callmangement.EHR.adapter.CampUploadedImagesListingAdapter;
import com.callmangement.databinding.ActivityViewCampDetailsBinding;
import com.callmangement.EHR.models.CampDetailsInfo;
import com.callmangement.EHR.models.CampDocInfo;
import com.callmangement.EHR.support.EqualSpacingItemDecoration;
import com.callmangement.EHR.support.Preference;
import com.callmangement.utils.PrefManager;

import java.util.ArrayList;

public class ViewCampDetailsActivity extends BaseActivity {

    Activity mActivity;
    private ActivityViewCampDetailsBinding binding;

    PrefManager preference;
    ArrayList<CampDocInfo> campDocInfoArrayList = null;
    private CampUploadedImagesListingAdapter adapter;

    CampDetailsInfo campDetailsInfo = null;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_FULLSCREEN);
        binding = ActivityViewCampDetailsBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        init();
    }

    private void init() {
        mActivity = this;
        preference =  new PrefManager(mActivity);

        Bundle bundle = getIntent().getExtras();
        campDocInfoArrayList = (ArrayList<CampDocInfo>) bundle.getSerializable("mylist");
        campDetailsInfo = (CampDetailsInfo) bundle.getSerializable("campDetails");

        binding.actionBar.ivBack.setVisibility(View.VISIBLE);
        binding.actionBar.ivThreeDot.setVisibility(View.GONE);
        binding.actionBar.layoutLanguage.setVisibility(View.GONE);
        binding.actionBar.buttonPDF.setVisibility(View.GONE);
        binding.actionBar.textToolbarTitle.setText(getResources().getString(R.string.view_camp_details));

        setUpData();
        setClickListener();
    }

    private void setUpData() {

        if (campDocInfoArrayList != null && campDocInfoArrayList.size() > 0) {
            binding.txtViewUploadedImages.setVisibility(View.VISIBLE);
            binding.rvAllCamps.setVisibility(View.VISIBLE);
        } else {
            binding.txtViewUploadedImages.setVisibility(View.GONE);
            binding.txtViewUploadedImages.setVisibility(View.GONE);
        }

        adapter = new CampUploadedImagesListingAdapter(mActivity, campDocInfoArrayList, onItemViewClickListener);
        binding.rvAllCamps.setHasFixedSize(true);
        GridLayoutManager layoutManager = new GridLayoutManager(mActivity, 3);
        binding.rvAllCamps.setLayoutManager(layoutManager);
        binding.rvAllCamps.addItemDecoration(new EqualSpacingItemDecoration(30, EqualSpacingItemDecoration.VERTICAL));
        binding.rvAllCamps.setAdapter(adapter);

        if (campDetailsInfo != null && campDetailsInfo.getStartDate() != null)
            binding.inputStartDateTime.setText(campDetailsInfo.getStartDate());

        if (campDetailsInfo != null && campDetailsInfo.getEndDate() != null)
            binding.inputEndDateTime.setText(campDetailsInfo.getEndDate());

        if (campDetailsInfo != null && campDetailsInfo.getOrganizeDate() != null)
            binding.inputOrganiseDateTime.setText(campDetailsInfo.getOrganizeDate());

        if (campDetailsInfo != null && campDetailsInfo.getDistrictNameEng() != null)
            binding.inputDistrict.setText(campDetailsInfo.getDistrictNameEng());

        if (campDetailsInfo != null && campDetailsInfo.getBlockName() != null)
            binding.inputBlock.setText(campDetailsInfo.getBlockName().toString());

        if (campDetailsInfo != null && campDetailsInfo.getAddress() != null)
            binding.inputPlace.setText(campDetailsInfo.getAddress());

        if (campDetailsInfo != null && campDetailsInfo.getDescription() != null)
            binding.inputDescription.setText(campDetailsInfo.getDescription());

    }

    CampUploadedImagesListingAdapter.OnItemViewClickListener onItemViewClickListener = new CampUploadedImagesListingAdapter.OnItemViewClickListener() {
        @Override
        public void onItemClick(CampDocInfo campDocInfo, int position) {
            startActivity(new Intent(mActivity, ZoomInZoomOutActivity.class).putExtra("image", campDocInfo.getDocumentPath()));
        }
    };

    private void setClickListener() {
        binding.actionBar.ivBack.setOnClickListener(view -> onBackPressed());
    }

}

File: app/src/main/java/com/callmangement/ehr/ehrActivities/ViewCampDetailsActivity.kt
package com.callmangement.ehr.ehrActivities

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.view.Window
import android.view.WindowManager
import androidx.recyclerview.widget.GridLayoutManager
import com.callmangement.R
import com.callmangement.databinding.ActivityViewCampDetailsBinding
import com.callmangement.ehr.adapter.CampUploadedImagesListingAdapter
import com.callmangement.ehr.models.CampDetailsInfo
import com.callmangement.ehr.models.CampDocInfo
import com.callmangement.ehr.support.EqualSpacingItemDecoration
import com.callmangement.utils.PrefManager

class ViewCampDetailsActivity : BaseActivity() {
    private var mActivity: Activity? = null
    private var binding: ActivityViewCampDetailsBinding? = null
    private var preference: PrefManager? = null
    private var campDocInfoArrayList: ArrayList<CampDocInfo>? = null
    private var adapter: CampUploadedImagesListingAdapter? = null
    private var campDetailsInfo: CampDetailsInfo? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        window.setFlags(
            WindowManager.LayoutParams.FLAG_FULLSCREEN,
            WindowManager.LayoutParams.FLAG_FULLSCREEN
        )
        binding = ActivityViewCampDetailsBinding.inflate(
            layoutInflater
        )
        setContentView(binding!!.root)
        init()
    }

    private fun init() {
        mActivity = this
        preference = PrefManager(mActivity!!)

        val bundle = intent.extras
        campDocInfoArrayList = bundle!!.getSerializable("mylist") as ArrayList<CampDocInfo>?
        campDetailsInfo = bundle.getSerializable("campDetails") as CampDetailsInfo?

        binding!!.actionBar.ivBack.visibility = View.VISIBLE
        binding!!.actionBar.ivThreeDot.visibility = View.GONE
        binding!!.actionBar.layoutLanguage.visibility = View.GONE
        binding!!.actionBar.buttonPDF.visibility = View.GONE
        binding!!.actionBar.textToolbarTitle.text = resources.getString(R.string.view_camp_details)

        setUpData()
        setClickListener()
    }

    private fun setUpData() {
        if (campDocInfoArrayList != null && campDocInfoArrayList!!.size > 0) {
            binding!!.txtViewUploadedImages.visibility = View.VISIBLE
            binding!!.rvAllCamps.visibility = View.VISIBLE
        } else {
            binding!!.txtViewUploadedImages.visibility = View.GONE
            binding!!.txtViewUploadedImages.visibility = View.GONE
        }

        adapter = CampUploadedImagesListingAdapter(
            mActivity!!,
            campDocInfoArrayList,
            onItemViewClickListener
        )
        binding!!.rvAllCamps.setHasFixedSize(true)
        val layoutManager = GridLayoutManager(mActivity, 3)
        binding!!.rvAllCamps.layoutManager = layoutManager
        binding!!.rvAllCamps.addItemDecoration(
            EqualSpacingItemDecoration(
                30,
                EqualSpacingItemDecoration.VERTICAL
            )
        )
        binding!!.rvAllCamps.adapter = adapter

        if (campDetailsInfo != null && campDetailsInfo!!.startDate != null) binding!!.inputStartDateTime.text =
            campDetailsInfo!!.startDate

        if (campDetailsInfo != null && campDetailsInfo!!.endDate != null) binding!!.inputEndDateTime.text =
            campDetailsInfo!!.endDate

        if (campDetailsInfo != null && campDetailsInfo!!.organizeDate != null) binding!!.inputOrganiseDateTime.text =
            campDetailsInfo!!.organizeDate

        if (campDetailsInfo != null && campDetailsInfo!!.districtNameEng != null) binding!!.inputDistrict.text =
            campDetailsInfo!!.districtNameEng

        if (campDetailsInfo != null && campDetailsInfo!!.blockName != null) binding!!.inputBlock.text =
            campDetailsInfo!!.blockName.toString()

        if (campDetailsInfo != null && campDetailsInfo!!.address != null) binding!!.inputPlace.text =
            campDetailsInfo!!.address

        if (campDetailsInfo != null && campDetailsInfo!!.description != null) binding!!.inputDescription.text =
            campDetailsInfo!!.description
    }

    var onItemViewClickListener: CampUploadedImagesListingAdapter.OnItemViewClickListener =
        object : CampUploadedImagesListingAdapter.OnItemViewClickListener {
            override fun onItemClick(campDocInfo: CampDocInfo?, position: Int) {
                startActivity(
                    Intent(mActivity, ZoomInZoomOutActivity::class.java).putExtra(
                        "image",
                        campDocInfo!!.documentPath
                    )
                )
            }
        }

    private fun setClickListener() {
        binding!!.actionBar.ivBack.setOnClickListener { view: View? -> onBackPressed() }
    }
}